libusb.go

```diff
@@ -133,7 +140,7 @@ func (ep libusbEndpoint) endpointDesc(dev *DeviceDesc) EndpointDesc {
 // and occasionally on convenience data types (like TransferType or DeviceDesc).
 type libusbIntf interface {
 	// context
-	init() (*libusbContext, error)
+	init(flags ...libusbOpt) (*libusbContext, error)
```

Comment:
I don't think this is the best approach. Keep the init() function as is. Instead, store the options inside the libusbImpl struct.